feat(flags): Document self-serve feature flags integrations (#12203)
getsentry/sentry-docs: docs/platforms/python/integrations/self-serve/index.mdx

---

> title: Self Serve Feature Flags

Reviewer: ```suggestion
title: Self-Serve Feature Flags
```

> title: Self-Serve Feature Flags

Reviewer 2: This is duplicated from https://github.com/getsentry/sentry-docs/pull/12152. Maybe I can close that one to make changes easier to coordinate?

description: "Learn how to attach custom feature flag data to Sentry error events."
---

<PlatformContent includePath="feature-flags/prerelease-alert" />

The Feature Flags integration allows you to manually track feature flag evaluations through an API. These evaluations are held in memory, and in the event an error occurs, sent to Sentry for review and analysis.
**At the moment, we only support boolean flag evaluations.**

## Install

Install `sentry-sdk` from PyPI.

```bash
pip install --upgrade 'sentry-sdk'
```

## Configure

Add `FeatureFlagsIntegration()` to your `integrations` list:

```python
import sentry_sdk

> from sentry_sdk.integrations.featureflags import FeatureFlagsIntegration

Reviewer 2: ```suggestion
from sentry_sdk.integrations.feature_flags import FeatureFlagsIntegration
```
After https://github.com/getsentry/sentry-python/pull/3902

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",
    integrations=[
        FeatureFlagsIntegration(),
    ],
)
```

## Verify

The integration is tested by calling the `add_feature_flag` API before capturing an exception.

```python
import sentry_sdk
from sentry_sdk.integrations.featureflags import add_feature_flag

add_feature_flag('test-flag', False)

sentry_sdk.capture_exception(Exception("Something went wrong!"))
```

> Visit the Sentry website and confirm that your error event has recorded the feature flag "test-flag" and its value "false".

Reviewer: ```suggestion
Go to your Sentry project and confirm that your error event has recorded the feature flag "test-flag" and its value "false".
```
